src/bridge/EspressoTEEVerifier.sol

```diff
@@ -129,16 +126,24 @@ contract EspressoTEEVerifier is IEspressoTEEVerifier, Ownable2Step {
     /*
      * @dev Set the mrEnclave of the contract
      */
-    function setMrEnclave(bytes32 _mrEnclave) external onlyOwner {
-        emit MREnclaveSet(_mrEnclave);
-        mrEnclave = _mrEnclave;
+    function setMrEnclave(bytes32 _mrEnclave, bool _isValid) external onlyOwner {
+        if (_isValid) {
+            emit MREnclaveSet(_mrEnclave);
```

Comment:
I think this should be emit after the value is set, we should probably also emit an event in the case that we are deleting an entry as well.